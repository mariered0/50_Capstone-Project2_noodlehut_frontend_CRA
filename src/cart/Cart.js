import React, { useContext, useState } from "react";
import cartContext from "./CartContext";
import CartItem from "./CartItem";
import CheckoutMessage from "./CheckoutMessage";
import { Box, Drawer, CssBaseline, Typography, Button, Divider } from "@mui/material";
import ShoppingCartCheckoutIcon from "@mui/icons-material/ShoppingCartCheckout";

const Cart = () => {
  
  const { cart, getTotalCost, isCartOpen, setIsCartOpen, deleteAllFromCart } =
    useContext(cartContext);
  const [isDialogOpen, setIsDialogOpen] = useState(false);

  return (
    <Box sx={{ display: "flex" }}>
      <CssBaseline />

      <Drawer
        sx={{
          width: 240,
          flexShrink: 0,
          "& .MuiDrawer-paper": {
            width: "30%",
          },
        }}
        variant="persistent"
        anchor="right"
        open={isCartOpen}
        onClose={() => setIsCartOpen(false)}
      >
        {/* Close & Clear button section */}
        <Box display="flex" justifyContent="right" m={2}>
          <Button
            variant="outlined"
            size="small"
            color="primary"
            onClick={() => setIsCartOpen(false)}
          >
            Close
          </Button>
          <Button
            variant="outlined"
            size="small"
            color="primary"
            onClick={() => deleteAllFromCart()}
          >
            Clear Cart
          </Button>
        </Box>

        <Box p={2} width="100%" textAlign="center" role="presentation">
          <Typography variant="h3" component="div">
            Your Cart
          </Typography>
          {cart?.map((item) => (
            <CartItem
              key={item.id}
              id={item.id}
              name={item.name}
              qty={item.quantity}
              price={item.price}
            />
          ))}
        </Box>

        <Box m={2}>
          <Divider variant="fullWidth" />
          <Typography variant="h3" align="right">
            Total: ${getTotalCost()}
          </Typography>
          <Box justifyContent="right" display="flex" m={1}>
              <Button
                variant="contained"
                size="large"
                startIcon={<ShoppingCartCheckoutIcon />}
                onClick={() => {setIsDialogOpen(true)}}
              >
                Go to Checkout
              </Button>
          </Box>
        </Box>
      </Drawer>

      {isDialogOpen ? <CheckoutMessage isDialogOpen={isDialogOpen} setIsDialogOpen={setIsDialogOpen} /> : ""}
    </Box>
  );
};

export default Cart;
