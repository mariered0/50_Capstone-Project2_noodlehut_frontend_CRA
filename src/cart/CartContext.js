import React from "react";

const cartContext = React.createContext();

export default cartContext;
cartContext.displayName = "Cart";