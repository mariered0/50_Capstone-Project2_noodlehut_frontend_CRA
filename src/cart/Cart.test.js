import React from "react";
import { render } from '@testing-library/react';
import CartContext from "./CartContext";
import Cart from "./Cart";
const { getTotalCost, isCartOpen, setIsCartOpen, deleteAllFromCart } = require("./_testCommon");
const cart = require("./_testCommon");


it("renders without crashing", () => {
    render(<Cart />);
});

it("matches snapshot", () => {
	const { asFragment } = render(
        <CartContext.Provider value={{ getTotalCost, cart, isCartOpen, setIsCartOpen, deleteAllFromCart }}>
            <Cart />
        </CartContext.Provider>
    
    );
	expect(asFragment()).toMatchSnapshot();
});

