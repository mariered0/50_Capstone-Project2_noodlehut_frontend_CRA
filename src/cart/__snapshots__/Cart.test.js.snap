// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-k008qs"
  >
    <div
      class="MuiDrawer-root MuiDrawer-docked css-1hnz2to-MuiDrawer-docked"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight css-18sg6k4-MuiPaper-root-MuiDrawer-paper"
      >
        <div
          class="MuiBox-root css-rl38ki"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Close
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Clear Cart
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-vrk4dv"
          role="presentation"
        >
          <div
            class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
          >
            Your Cart
          </div>
          <div
            class="MuiBox-root css-i9gxme"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-gy4mli-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="image_unavailable.png"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-1lv9nvb-MuiTypography-root"
                >
                   Fried Tofu
                </h6>
                <div
                  class="MuiBox-root css-gg4vpm"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    1 total
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    $6.95
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
              >
                <div
                  class="MuiStack-root css-nen11g-MuiStack-root"
                >
                  <button
                    aria-label="add one"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AddCircleOutlineIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-label="delete one"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="RemoveCircleOutlineIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-15oeajw"
                >
                  <button
                    aria-label="delete"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CloseIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-inset css-uls1yk-MuiDivider-root"
            />
          </div>
          <div
            class="MuiBox-root css-i9gxme"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-gy4mli-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="image_unavailable.png"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-1lv9nvb-MuiTypography-root"
                >
                   Spring Rolls
                </h6>
                <div
                  class="MuiBox-root css-gg4vpm"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    1 total
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    $6.95
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
              >
                <div
                  class="MuiStack-root css-nen11g-MuiStack-root"
                >
                  <button
                    aria-label="add one"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AddCircleOutlineIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-label="delete one"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="RemoveCircleOutlineIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-15oeajw"
                >
                  <button
                    aria-label="delete"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CloseIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-inset css-uls1yk-MuiDivider-root"
            />
          </div>
          <div
            class="MuiBox-root css-i9gxme"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-gy4mli-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="image_unavailable.png"
                  />
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-uqwprf-MuiGrid-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft css-1lv9nvb-MuiTypography-root"
                >
                   Pot Sticker
                </h6>
                <div
                  class="MuiBox-root css-gg4vpm"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    5 total
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    $6.95
                  </p>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
              >
                <div
                  class="MuiStack-root css-nen11g-MuiStack-root"
                >
                  <button
                    aria-label="add one"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AddCircleOutlineIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-label="delete one"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="RemoveCircleOutlineIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-15oeajw"
                >
                  <button
                    aria-label="delete"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CloseIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-inset css-uls1yk-MuiDivider-root"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-1c5ij41"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-alignRight css-5skc57-MuiTypography-root"
          >
            Total: $
          </h3>
          <div
            class="MuiBox-root css-quj821"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ShoppingCartCheckoutIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03L21 4.96 19.25 4l-3.7 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2zM12 2l4 4-4 4-1.41-1.41L12.17 7H8V5h4.17l-1.59-1.59L12 2z"
                  />
                </svg>
              </span>
              Go to Checkout
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
